administration/src/bp-modules/components/CustomDatePicker.tsx

```diff
@@ -1,62 +1,88 @@
-  date: Date | null
-  onBlur: () => void
-  onChange: (date: Date | null) => void
-  onClear: () => void
+  value?: Date | null
+  disabled?: boolean
+  label?: string
+  onBlur?: () => void
+  onChange?: (date: Date | null) => void
+  onClear?: () => void
   isValid: boolean
   minDate?: Date
   maxDate?: Date
-  disableFuture: boolean
+  disableFuture?: boolean
+  disablePast?: boolean
+  textFieldHelperText?: string
+  textFieldSlotProps?: DesktopDatePickerSlotProps<Date, true>['textField']
 }
 
 const CustomDatePicker = ({
-  date,
+  value,
+  disabled,
+  label,
   onBlur,
   onChange,
   onClear,
   isValid,
   minDate,
   maxDate,
   disableFuture,
-}: CustomDatePickerProps): ReactElement => {
-  const { viewportSmall } = useWindowDimensions()
-  const formStyle = viewportSmall ? { fontSize: 16, padding: '9px 10px' } : { fontSize: 14, padding: '6px 10px' }
-  const textFieldBoxShadow =
-    '0 0 0 0 rgba(205, 66, 70, 0), 0 0 0 0 rgba(205, 66, 70, 0), inset 0 0 0 1px #cd4246, inset 0 0 0 1px rgba(17, 20, 24, 0.2), inset 0 1px 1px rgba(17, 20, 24, 0.3)'
-  return (
-    <DesktopDatePicker
-      views={['year', 'month', 'day']}
-      value={date}
-      format='dd.MM.yyyy'
-      slotProps={{
-        clearIcon: { fontSize: viewportSmall ? 'medium' : 'small' },
-        openPickerIcon: { fontSize: 'small' },
-        field: { clearable: true, onClear },
-        textField: {
+  disablePast,
+  textFieldHelperText,
+  textFieldSlotProps,
+}: CustomDatePickerProps): ReactElement => (
+  <DesktopDatePicker
+    disabled={disabled}
+    label={label}
+    views={['year', 'month', 'day']}
+    openTo='year'
+    value={value}
+    format='dd.MM.yyyy'
+    sx={{ '& input[value=""]:not(:focus)': { color: 'transparent' } }}
+    slotProps={{
+      openPickerIcon: { fontSize: 'small' },
+      field: {
+        clearable: true,
+        onClear,
+      },
+      textField: deepmerge(
+        {
+          helperText: textFieldHelperText,
           placeholder: 'TT.MM.JJJJ',
           error: !isValid,
           spellCheck: false,
           onBlur,
+          size: 'small',
           sx: {
             width: '100%',
-            input: formStyle,
-            boxShadow: !isValid ? textFieldBoxShadow : undefined,
+            boxShadow: !isValid
+              ? `
+                0 0 0 0 rgba(205, 66, 70, 0),
+                inset 0 0 0 1px #cd4246,
+                inset 0 0 0 1px rgba(17, 20, 24, 0.2),
+                inset 0 1px 1px rgba(17, 20, 24, 0.3),
+              `
+              : undefined,
+            '.MuiPickersInputBase-root': {
+              borderRadius: '2px',
+            },
           },
         },
-      }}
-      localeText={deDE.components.MuiLocalizationProvider.defaultProps.localeText}
-      disableFuture={disableFuture}
-      minDate={minDate}
-      maxDate={maxDate}
-      onChange={onChange}
-    />
-  )
-}
+        textFieldSlotProps
+      ),
+    }}
+    enableAccessibleFieldDOMStructure
```

Comment:
can you please enable this only for viewport small @andrew8er 
Not be able to escape the field with keyboard is bad a11y

Reply:
MUI switches between the desktop and mobile variant based on the `@media (pointer: fine)` query, as explained [here](https://mui.com/x/react-date-pickers/date-picker/#available-components). I think it would be more consistent to use this query here also then.

Reply:
I'm open for any solutions as long as I can leave the field with keyboard :)